### PR TITLE
[Enhancement] Support specifying AnonymousAWSCredentialsProvider after removing the broker in ingestion

### LINKED ISSUE #15548: [Enhancement] Support AnonymousAWSCredentialsProvider for broker load without broker

### PR DESCRIPTION
**Pull request by xiaoyong-z**
Support specifying AnonymousAWSCredentialsProvider after removing the broker in ingestion.
According to doc link https://hadoop.apache.org/docs/stable/hadoop-aws/tools/hadoop-aws/index.html, several default providers are chained, it should suffice for most cases.
```
    <property>
        <name>fs.s3a.aws.credentials.provider</name>
        <value>
            org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider,
            org.apache.hadoop.fs.s3a.TemporaryAWSCredentialsProvider,
            org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider,
            com.amazonaws.auth.EnvironmentVariableCredentialsProvider,
            org.apache.hadoop.fs.s3a.auth.IAMInstanceCredentialsProvider
            com.amazonaws.auth.InstanceProfileCredentialsProvider
        </value>
    </property>
```
For org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider, it doesn't need ak/sk, here we remove the check for ak/sk for broker load, following is the example for org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider
```
LOAD LABEL test.label
(
    DATA INFILE("xxxxx")
    INTO TABLE tab
    FORMAT AS "csv"
)
WITH BROKER
(
    "fs.s3a.endpoint" = "s3-us-west-2.amazonaws.com",
    "fs.s3a.aws.credentials.provider" = "org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider"
);
```
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #15548

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsFsManager.java [hunk 1]
```diff
@@ -63,15 +64,25 @@ class ConfigurationWrap extends Configuration {
...
             if (regionId.contains("-internal")) {
                 return regionId.substring(0, regionId.length() - "-internal".length());
             } else {
                 return regionId;
             }
+        } else if (tObjectStoreType == TObjectStoreType.KS3) {
+            String[] hostSplit = endPoint.split("\\.");
+            String regionId = hostSplit[0];
+            if (regionId.contains("-internal")) {
+                return regionId.substring(4, regionId.length() - "-internal".length() - 4);
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsFsManager.java:82]: What's the meaning of internal?

Comment (xiaoyong-z) [reply fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsFsManager.java:82]: for oss and ks3, it has extranet endpoint and intranet endpint, 
extranet endpoint will have the 'internal'. 
Take the OSS for example:
```
extranet endpoint: oss-cn-hangzhou.aliyuncs.com 
intranet endpoint: oss-cn-hangzhou-internal.aliyuncs.com
region-id: oss-cn-hangzhou
```
when parsing the region from endpoint, we need to remove the 'internal' here

**Review by chaoyli — DISMISSED** [review]

**Review by meegoo — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
@mergify rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 17 (29.41%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|5|17|29.41%|[70, 78, 79, 80, 81, 82, 84, 133, 153, 173, 193, 537]|

**Review by meegoo — DISMISSED** [review]

**Review by meegoo — DISMISSED** [review]

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by chaoyli** [comment]
Support specifying AnonymousAWSCredentialsProvider after removing the broker in ingestion.
According to doc link https://hadoop.apache.org/docs/stable/hadoop-aws/tools/hadoop-aws/index.html, several default providers are chained, it should suffice for most cases.
```
    <property>
        <name>fs.s3a.aws.credentials.provider</name>
        <value>
            org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider,
            org.apache.hadoop.fs.s3a.TemporaryAWSCredentialsProvider,
            org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider,
            com.amazonaws.auth.EnvironmentVariableCredentialsProvider,
            org.apache.hadoop.fs.s3a.auth.IAMInstanceCredentialsProvider
            com.amazonaws.auth.InstanceProfileCredentialsProvider
        </value>
    </property>
```
For org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider, it doesn't need ak/sk, here we remove the check for ak/sk for broker load, following is the example for org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider
```
LOAD LABEL test.label
(
    DATA INFILE("xxxxx")
    INTO TABLE tab
    FORMAT AS "csv"
)
WITH BROKER
(
    "fs.s3a.endpoint" = "s3-us-west-2.amazonaws.com",
    "fs.s3a.aws.credentials.provider" = "org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider"
);
```
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #15548

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by xiaoyong-z** [comment]
@mergify rebase

**Comment by xiaoyong-z** [comment]
@mergify rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 17 (29.41%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|5|17|29.41%|[73, 81, 82, 83, 84, 85, 87, 136, 156, 176, 196, 563]|

**Comment by xiaoyong-z** [comment]
@mergify rebase

**Comment by xiaoyong-z** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 17 (29.41%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|5|17|29.41%|[73, 81, 82, 83, 84, 85, 87, 136, 156, 176, 196, 563]|

**Comment by xiaoyong-z** [comment]
run starrocks_be_unittest image

**Comment by wanpengfei-git** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_be_unittest

**Review by meegoo — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**